web/__tests__/components/Feed.unit.test.tsx

```diff
@@ -0,0 +1,110 @@
+import { getPrefFilters, Option, SelectedFilters } from "../../components/Feed/Feed";
+import { Role } from "../../utils/types/account";
+import { Breed, FosterType, Status } from "../../utils/types/post";
+import { IUser } from "../../utils/types/user";
+
+// Mock authentication
+jest.mock("../../context/auth", () => ({
+  useAuth: jest.fn()
+}));
+
+// Function to count the number of filter options
+const countFilters = (filters: SelectedFilters | null): number => {
+  if (!filters) return 0;
+
+  let count = 0;
+  for (const opts of Object.values(filters)) {
+    if (!opts) continue;
+    count = Array.isArray(opts) ? count + opts.length : count + 1;
+  }
+  return count;
+};
+
+// Function to determine whether two options are equivalent
+const isEqual = (a: Option, b: Option): boolean => (
+  a.value === b.value && a.label === b.label
+);
+
+// Function to 
+const checkFilters = (filters: SelectedFilters | null, target: SelectedFilters): boolean => {
+  if (!filters) return false;
+  return Object.keys(filters).filter((filter) => filter in target).every((f) => (
+    filters[f].every((opt) => target[f].some((o) => isEqual(o, opt))
+    && filters[f].length === target[f].length
+  )));
+}
+
+/**
+ * Tests ability to use user preferences to propagate feed filters
+ */
```

Comment:
Add `@group` tags like in `api/account/account.unit.test.ts` - it allows us to run groups of tests. The groups should be:

```
components/feed
components
unit
```